src/main/java/refactoringml/RefactoringAnalyzer.java

```diff
@@ -42,102 +76,129 @@ public RefactoringAnalyzer (Project project, Database db, Repository repository,
 					 */
 					String refactoredClassNameFromRMiner = pair.getRight();
-					if(isAnonymousClass(refactoredClassNameFromRMiner)) {
-						log.error("Refactoring in an anonymous class, which we skip: " + refactoredClassNameFromRMiner + ", commit = " + superCommitMetaData + ", refactoring = " + shortSummary(refactoringSummary));
+					if (isAnonymousClass(refactoredClassNameFromRMiner)) {
+						String shortSummary = shortSummary(refactoringSummary);
+						log.error(
+								"Refactoring in an anonymous class, which we skip: {}, commit = {}, , refactoring = {}",
+								refactoredClassNameFromRMiner, superCommitMetaData, shortSummary);
 						continue;
 					}
-					if(!persistedCommitMetaData){
+					if (!persistedCommitMetaData) {
 						persistedCommitMetaData = true;
-						db.persist(superCommitMetaData);
+						db.save(superCommitMetaData);
+						toRemoveOnException.add(superCommitMetaData);
+
 					}
-					ImmutablePair<String, String> refactoredClassName = new ImmutablePair<>(refactoredClassNameFromRMiner, classAliases.get(refactoredClassNameFromRMiner));
+					ImmutablePair<String, String> refactoredClassName = new ImmutablePair<>(
+							refactoredClassNameFromRMiner, classAliases.get(refactoredClassNameFromRMiner));
 
 					// build the full RefactoringCommit object
-					RefactoringCommit refactoringCommit = buildRefactoringCommitObject(superCommitMetaData, refactoring, refactoringSummary, refactoredClassName, refactoredClassFile);
-
-					if (refactoringCommit != null) {
+					List<RefactoringCommit> commits = buildRefactoringCommitObject(superCommitMetaData, refactoring,
+							refactoringSummary, refactoredClassName, refactoredClassFile);
+					toRemoveOnException.addAll(commits);
+					if (!commits.isEmpty()) {
+						RefactoringCommit refactoringCommit = commits.get(0);
 						// mark it for the process metrics collection
-						allRefactorings.add(refactoringCommit);
+						allRefactorings.add(commits.get(0));
 
-						if(storeFullSourceCode)
+						if (storeFullSourceCode)
 							storeSourceCode(refactoringCommit.getId(), refactoring, commit);
 					} else {
-						log.debug("RefactoringCommit instance was not created for the class: " + refactoredClassName + " and the refactoring type: " + refactoring.getName()  + " on commit " + commit.getName());
+						log.debug(
+								"RefactoringCommit instance was not created for the class: {} and the refactoring type: {} on commit {}",
+								refactoredClassName, refactoring.getName(), commit.getName());
 					}
 				}
 			}
-		} catch (Exception e){
-			log.error("Failed to collect commit data for a refactored commit." + createErrorState(superCommitMetaData.getCommitId(), project), e);
+		} catch (Exception e) {
+			log.error("Failed to collect commit data for a refactored commit. {}",
+					createErrorState(superCommitMetaData.getCommitId(), project), e);
+			toRemoveOnException.forEach(db::deleteWithoutThrowingOnException);
```

Comment:
Why did you add this list(toRemoveOnException)? In case of an exception, we are rerolling the last transaction (the last commit) in App.java anyways.